sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -930,7 +930,24 @@ private class BulkWriter(container: CosmosAsyncContainer,
                   activeReadManyOperationsSnapshot,
                   numberOfIntervalsWithIdenticalActiveOperationSnapshots
                 )
+
+                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get == 1L) {
+                  val activeWriteOperationsSnapshot = activeBulkWriteOperations.clone()
+                  activeWriteOperationsSnapshot.foreach(operation => {
+                    if (activeBulkWriteOperations.contains(operation)) {
+                      // re-validating whether the operation is still active - if so, just re-enqueue another retry
+                      // this is harmless - because all bulkItemOperations from Spark connector are always idempotent
+
+                      // For FAIL_NON_SERIALIZED, will keep retry, while for other errors, use the default behavior
+                      bulkInputEmitter.emitNext(operation, emitFailureHandler)
```

Comment:
for ItemBulkUpdate -> we might need to change here to emit from the readManyEmitter instead depending on here the hang operation happens

(do not think need to change in this PR though)

Reply:
Yeah - initially left read-many out intentionally because we had not seen a hang there. But it won't be hard - so, better address it in this PR. Please take a look at the latest version.